### PR TITLE
refactor: adapt checkout shipping page with formly 

### PR DESCRIPTION
**Pull request by MaxKless**
<br /><br />## PR Type<br /><br /><br /><br />[ ] Bugfix<br />[ ] Feature<br />[ ] Code style update (formatting, local variables)<br />[x] Refactoring (no functional changes, no API changes)<br />[ ] Build-related changes<br />[ ] CI-related changes<br />[ ] Documentation content changes<br />[ ] Application / infrastructure changes<br />[ ] Other: <br /><br />## What Is the Current Behavior?<br /><br /><br /><br />Issue Number: Closes #<br /><br />## What Is the New Behavior?<br /><br />Checkout shipping page now uses formly and is reworked. No longer conforms to a now-useless container-component pattern.<br /><br />## Todo<br />- [ ] fix failing checkout shipping page component test <br />- [ ] !!!! re-enable skipped test !!!!<br />- [x] split submittedSubject test<br />- [ ] facade methods without fetch<br /><br />## Does this PR Introduce a Breaking Change?<br /><br /><br /><br />[ ] Yes<br />[x] No<br /><br />## Other Information<br />

[AB#64843](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/64843)

### CODE REVIEW

**Comment by SGrueber** [comment]
Automatic preselection of a shipping method is not working any more:
- login and add products to basket, checkout
- on checkout address page add an Bulgarian address
- continue checkout
- > on checkout shipping page the international express delivery is displayed but not selected

**Comment by SGrueber** [comment]
image
 
Please remove the 'b' :-)

**Review by SGrueber — CHANGES_REQUESTED** [review]
Under certain circumstances 2 patch requests are send to the sever after the shipping method has been changed:
- first patch request is fine
- second path request sets the shipping method to the first value

imo, the many observables of the checkout-shipping.component are too dynamic and they influence each other, so the code is really hard to understand.
I assume after changing the shipping method the basket is recalculated and retrieved from the server, that triggers the setupShippingMethodPreselection() functionality.

File: src/app/pages/checkout-shipping/checkout-shipping-page.component.html [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+        <div *ngIf="shippingMethods.length; else noShippingMethods" class="shipping-methods">
+          <div *ngIf="nextDisabled && !(basketError$ | async)" role="alert" class="alert alert-danger">
+            {{ 'checkout.shipping_method.no_Selection.error' | translate }}
+          </div>
+          <h3>{{ 'checkout.shipping_method.selection.heading' | translate }}</h3>
+          <ish-checkout-shipping></ish-checkout-shipping>
+        </div>
+
+        <ng-template #noShippingMethods>
+          b
```
[the remark this answers is not in the document]
Comment (MaxKless) [reply src/app/pages/checkout-shipping/checkout-shipping-page.component.html:22]: haha woops

File: src/app/pages/checkout-shipping/checkout-shipping-page.component.ts [hunk 2]
```diff
@@ -11,29 +13,43 @@ import { ShippingMethod } from 'ish-core/models/shipping-method/shipping-method.
...
-
-  updateBasketShippingMethod(shippingId: string) {
-    this.checkoutFacade.updateBasketShippingMethod(shippingId);
+    this.shippingMethods$ = this.checkoutFacade.eligibleShippingMethods$();
+    this.isBusinessCustomer$ = this.accountFacade.isBusinessCustomer$;
   }
 
   /**
-   * Validates the basket and jumps to the next checkout step (Payment)
+   * leads to next checkout page (checkout payment)
```
[the remark this answers is not in the document]
Comment (MaxKless) [reply src/app/pages/checkout-shipping/checkout-shipping-page.component.ts:38]: not sure where that came from, I don't remember changing it. Fixed.

**Pull request merged**